protocol_generator/protocol_generator.cpp

```diff
@@ -0,0 +1,234 @@
+		for (float max_dose = 0.5; max_dose < 10.25; max_dose += 0.25) {
+			max_doses.push_back(max_dose);
+		}
+		s.push({ target, {}, max_doses });
+		std::vector<std::vector<float>> chosen_protocols;
+		while (!s.empty()) {
+			DataFrame top = s.top();
+			s.pop(); 
+			if (top.sum < 0) continue;
+			if (top.sum == 0) {
+				std::cout << target << ":" << ways << ": ";
+				for (int i = 0; i < top.coins.size(); i++) {
+					std::cout << top.coins[i] << " ";
+				}
+				chosen_protocols.push_back(top.coins);
+				std::cout << "\n";
+				++ways;
+				continue;
+			}
+			if (top.avail_coins.empty()) continue;
+			DataFrame d = top;
+			d.sum -= top.avail_coins[0];
+			d.coins.push_back(top.avail_coins[0]);
+			s.push(d);
+			d = top;
+			d.avail_coins.erase(std::begin(d.avail_coins));
+			s.push(d);
+		}
+		std::cout << target << ":" << ways << std::endl;
+		protocol_map[target] = chosen_protocols;
+	}
+	std::cout << ways << std::endl;
+	std::cout << "Size " << protocol_map.size() << "\n";
+  	std::cout << "Count " << protocol_map.count(2.0) << "\n";
+
+	// 1. Losowanko
+	// 1.1 losuj sumę - równa dystrybucja po wartości sumy
+	// 1.2 losuj pozycję
+	// 1.3 losuj permutacje
+	// 1.4 losuj czasy
+
+	std::set<std::vector<float>> protocol_set;
+
+	std::random_device rd;
+	std::mt19937 rng;
+	rng.seed(rd());
+	std::uniform_int_distribution<int> dist_sum(3, 40); 
+	int i = 0;
+	std::ofstream f("protocol_list_1.csv");
+	std::ofstream t("protocol_times_1.csv");
+	int file_count = 1;
+	while (i < 50000) {
+		// Ile linii na jeden plik
+		if (i > file_count * 1570) {
+			file_count++;
+			f.close();
+			t.close();
+			f.open("protocol_list_" + std::to_string(file_count) + ".csv");
+			t.open("protocol_times_" + std::to_string(file_count) + ".csv");
+		}
+		// Losuj sumę
+		float sum = dist_sum(rng) * 0.25;
+
+		std::vector<float> temp_protocol;
+
+		std::cout << "Suma " << sum << "\n";
+		std::cout << "Rozmiar " << protocol_map[sum].size() << "\n";
+		// Losuj pozycję
+		std::uniform_int_distribution<int> dist_pos(
+			0, protocol_map[sum].size() - 1);
+		int temp_pos = dist_pos(rng);
+		std::cout << "Poz " << temp_pos << "\n";
+
+		temp_protocol = (protocol_map[sum])[temp_pos];
+
+		int temp_length = temp_protocol.size();
+
+		std::cout << "Dlugosc " << temp_length << "\n";
+
+		std::sort(temp_protocol.begin(), temp_protocol.begin() + temp_length);
+
+		// Losuj permutację
+		std::random_shuffle ( temp_protocol.begin(), temp_protocol.end() );
+
+		// Losuj 2 czasy
+		// 72000 kroków symulacji co 6 sekund
+		// zaczynamy w pierwszym kroku nr 1
+		// między dawkami 30min = 300 kroków
+		// czyli:
+		// 1.1 weź długość dawek. dawkki mogą zacząć się najpóźniej 
+		//     w 72000 - długość * 300 kroku = 72000 - koniec
+		// 1.2 losuj pierwszy termin dawki z 1..72000-koniec
+		// 1.3 dodaj do niego 300 i losuj kolejny termin dawki z 
+		//     (poprzedni + 300, 14400-koniec + 300)
+		// 1.4 itd, aż nie dojdziesz do końca długości wektorów

```

Comment:
To akutalne? Czy to nie był ten sposób, który daje niejednostajny rozkład dawek?